ci: add mandatory LLM-involvement category disclosure (#2343)
nix-community/stylix: .github/PULL_REQUEST_TEMPLATE.md

drastically accelerate the reviewing and merging process of this PR.
-->
- [ ] <!-- MANDATORY --> I certify that I have the right to submit this contribution under the [MIT license](https://github.com/nix-community/stylix/blob/master/LICENSE)
- [ ] <!-- MANDATORY --> I have disclosed the single most severe LLM-involvement category that applies to this contribution:

>   - [ ] `llm: 0`: Manual: The absence of LLMs would result in equal results and quality
>   - [ ] `llm: 1`: Syntactic: Includes non-negligible LLM content

Reviewer: `llm: 0` is very open to interpretation, where as the other three are significantly more clear. I would prefer having `llm: 0` be non-existent LLM usage and `llm: 1` be any LLM usage, 

```suggestion
  - [ ] `llm: 0`: Manual: LLMs were absent in the creation of this PR
  - [ ] `llm: 1`: Syntactic: Includes LLM generated content
```

Author: > `llm: 0` is very open to interpretation, where as the other three are significantly more clear. I would prefer having `llm: 0` be non-existent LLM usage

It intentionally has a faithfull loophole inspired by:

> ```md
> These rules apply only to outside contributions to Ghostty. Maintainers
> are exempt from these rules and may use AI tools at their discretion;
> they've proven themselves trustworthy to apply good judgment.
> ```
>
> -- [github:ghostty-org/ghostty, `/AI_POLICY.md`](https://github.com/ghostty-org/ghostty/blob/6246c288ae1087c8d67f75432a59da004b30bf25/AI_POLICY.md?plain=1#L33-L35)

I suggest initially going with this, allowing competent contributors to faithfully disclose their LLM dependency because it seems well-regarded Nix maintainers use LLMs for minor things. Regardless of the label description, it is suspect to dishonesty that I suggest ignoring for now:

> Considering the inherent LLM discrimination introduced by the `13.llm: n` labels, malicious or dishonest contributors might intentionally miscategorize their contributions towards `13.llm: 0` to potentially avoid being ignored by LLM skeptics. Unless it becomes obvious that a non-negligible amount of contributions are intentionally miscategorized, I suggest faithfully assuming contributors to be honest, before designing a LLM ban procedure.
>
> -- https://github.com/NixOS/nixpkgs/issues/410741#issuecomment-3807405985

> and `llm: 1` be any LLM usage

This also intentionally has a smaller loophole, allowing trivial use cases like LLMs fixing trivial syntax issues or suggesting the most idiomatic function to use. I suppose what this really entails will be clear once we see how people interpret this.

I see your point of more explicitly disclosing LLM contributions, but I suggest going with this for now and later determining how much people abuse these faithfull definitions.

  - [ ] `llm: 2`: Semantic: LLM involved in design decisions or core logic

>   - [ ] `llm: 3`: Unaudited: LLM output with little to no human verification

Reviewer: We should make it clear that is level of LLM involvement is not welcome in Stylix, as was agreed on in https://github.com/nix-community/stylix/issues/2178.

Author: > We should make it clear that is level of LLM involvement is not welcome in Stylix, as was agreed on in #2178.

IMHO, the categorization is already fairly discriminatory towards LLM contributions, making it somewhat obvious that unaudited contributions are unwelcome. These are also rude submissions, but this is irrelevant to the LLM categorization.

I wonder whether we will even receive submissions categorized as unaudited. Keep in mind that you are free to ignore such submissions or only very superficially interact with them.
